src/m365/spo/commands/site/site-recyclebinitem-move.ts

```diff
@@ -0,0 +1,201 @@
+        ids: typeof args.options.ids !== 'undefined',
+        all: !!args.options.all,
+        confirm: !!args.options.confirm
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --siteUrl <siteUrl>'
+      },
+      {
+        option: '-i, --ids [ids]'
+      },
+      {
+        option: '--all'
+      },
+      {
+        option: '--confirm'
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['ids', 'all'] }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.siteUrl);
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.ids && !validation.isValidGuidArray(args.options.ids.split(','))) {
+          return 'ids contains invalid GUID';
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Moving items from the first-stage recycle bin to the second-stage recycle bin at ${args.options.siteUrl}...`);
+    }
+
+    if (args.options.confirm) {
+      await this.moveRecycleBinItem(args, logger);
+    }
+    else {
+      const result = await Cli.prompt<{ continue: boolean }>({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: 'Are you sure you want to move the items to the second-stage recycle bin?'
+      });
+
+      if (result.continue) {
+        await this.moveRecycleBinItem(args, logger);
+      }
+    }
+  }
+
+  private async moveRecycleBinItem(args: CommandArgs, logger: Logger): Promise<void> {
+    try {
+      if (args.options.all) {
+        if (this.verbose) {
+          logger.logToStderr('Moving all items to the second-stage recycle bin');
+        }
+        const requestOptions: CliRequestOptions = {
+          url: `${args.options.siteUrl}/_api/web/recycleBin/MoveAllToSecondStage`,
+          headers: {
+            'accept': 'application/json;odata=nometadata'
+          },
+          responseType: 'json'
+        };
+        const response = await request.post<{ 'odata.null': boolean }>(requestOptions);
+        if (!response['odata.null']) {
+          throw 'Something went wrong when moving the selected items to the second-stage recycle bin';
+        }
+      }
+      else {
+        if (this.verbose) {
+          logger.logToStderr(`Moving ${args.options.ids} to the second-stage recycle bin`);
+        }
+
+        await this.postBatch(args.options.ids!.split(','), logger, args.options.siteUrl);
+      }
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async postBatch(ids: string[], logger: Logger, siteUrl: string): Promise<void> {
```

Comment:
You are probably going to hate me, but after doing some research I found an API endpoint which is way much more easier. We should probably pick this request instead of a batch:

```http
POST https://contoso.sharepoint.com/sites/project-x/_api/site/RecycleBin/MoveToSecondStageByIds
{
  "ids": ["9dcd5c22-b895-40fb-8a9b-a93124674592","5312a14c-493e-4cec-98de-78d6585a7d5d"]
}
```

Reply:
![fuck my life](https://media.tenor.com/0P6u2cTN6fMAAAAM/shine-text-share-the-flip-side.gif)